### PR TITLE
core/cib: add peek capabilities

### PR DESCRIPTION
**Pull request by lebrush**
I needed that on a project I was working on...

### CODE REVIEW

File: core/include/cib.h [hunk 1]
```diff
@@ -87,6 +87,22 @@ static inline int cib_get(cib_t *__restrict cib)
 }
 
 /**
+ * @brief Get the index of the next item in buffer without removing it.
+ *
+ * @param[in,out] cib   corresponding *cib* to buffer.
+ *                      Must not be NULL.
+ * @return index of next item, -1 if the buffer is empty
+ */
+static inline int cib_peek(cib_t *__restrict cib)
```
Comment (gebart) [inline core/include/cib.h]: does `restrict` actually do anything when there's only one pointer involved?

**Comment by gebart** [comment]
ACK for the change.
My question about `restrict` is merely a curiosity, not something that has to be addressed in this PR because I saw that all functions in cib.h have the same qualifier.

**Comment by gebart** [comment]
One thing came up after I posted the above:

Please add some `cib_peek` usage in `tests/unittests/tests-core/tests-core-cib.c` to keep the test coverage high

Needs a second ACK.

**Comment by gebart** [comment]
unit test looks good, ACK from me, needs a second ACK. @kaspar030 @OlegHahm maybe?

**Comment by OlegHahm** [comment]
ACK and go!

**Pull request merged**